app/screens/convert_gm_to_channel/message_box/message_box.tsx

```diff
@@ -0,0 +1,138 @@
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Text, View} from 'react-native';
+
+import CompassIcon from '@app/components/compass_icon';
+import {useTheme} from '@context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type MessageBoxTypes = 'default' | 'danger'
+
+type Props = {
+    header: string;
+    body: string;
+    type?: MessageBoxTypes;
+}
+
+const getBaseStyles = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.sidebarTextActiveBorder, 0.08),
+            borderWidth: 1,
+            borderRadius: 8,
+            borderColor: changeOpacity(theme.sidebarTextActiveBorder, 0.16),
+            display: 'flex',
+            flexDirection: 'row',
+            padding: 16,
+            gap: 12,
+        },
+        icon: {
+            marginTop: 5,
+            fontSize: 20,
+            width: 28,
+            height: 28,
+            borderWidth: 3,
+            color: theme.sidebarTextActiveBorder,
+            borderColor: theme.sidebarTextActiveBorder,
+            borderRadius: 14,
+            textAlign: 'center',
+        },
+        iconContainer: {},
+        textContainer: {
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 8,
+            flex: 1,
+        },
+        heading: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        body: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+        },
+    };
+});
+
+const getDefaultStylesFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.sidebarTextActiveBorder, 0.08),
+            borderColor: changeOpacity(theme.sidebarTextActiveBorder, 0.16),
+        },
+        icon: {
+            color: theme.sidebarTextActiveBorder,
+            borderColor: theme.sidebarTextActiveBorder,
+        },
+    };
+});
+
+const getDangerStylesFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.dndIndicator, 0.08),
+            borderColor: changeOpacity(theme.dndIndicator, 0.16),
+        },
+        icon: {
+            color: theme.dndIndicator,
+            borderColor: theme.dndIndicator,
+        },
+    };
+});
+
+const getStyleFromTheme = (theme: Theme, kind: MessageBoxTypes | undefined) => {
+    let kindStyles;
+    switch (kind) {
+        case 'danger': {
+            kindStyles = getDangerStylesFromTheme(theme);
+            break;
+        }
+        default: {
+            kindStyles = getDefaultStylesFromTheme(theme);
+            break;
+        }
+    }
+
+    return {
+        ...kindStyles,
+    };
```

Comment:
If you do this, you are returning a new object on every call, defeiting the purpose of `makeStyleSheetFromTheme` (which tries to return always the same object).

In this context doesn't feel like there is much difference, since you are never checking equality over the whole style but the inner properties, that don't change in this context. Still, it feels weird.

Reply:
Yeah, it was left over from another review fix. Fixed now.